code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,315 @@
+		QDEL_NULL(current_beam)
+		return
+	current_beam = link_owner.beam(link_target, icon_state= "medbeam", beam_type = /obj/effect/ebeam/essence_link)
+	update_beam()
+
+/// Updates the link's appearance.
+/datum/status_effect/stacking/essence_link/proc/update_beam()
+	current_beam?.visuals.alpha = round(255 / (max_stacks+1 - stacks))
+
+// ***************************************
+// *********** Salve Regeneration
+// ***************************************
+/datum/status_effect/salve_regen
+	id = "salve regen"
+	duration = 15 SECONDS
+	tick_interval = 2 SECONDS
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = null
+	/// The owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+
+/datum/status_effect/salve_regen/on_apply()
+	buff_owner = owner
+	if(!isxeno(buff_owner))
+		return FALSE
+	buff_owner.balloon_alert(buff_owner, "Salve regeneration")
+	return TRUE
+
+/datum/status_effect/salve_regen/on_remove()
+	buff_owner.balloon_alert(buff_owner, "Salve regeneration ended")
+	return ..()
+
+/datum/status_effect/salve_regen/tick()
+	new /obj/effect/temp_visual/healing(get_turf(buff_owner))
+	var/heal_amount = buff_owner.maxHealth * 0.01
+	buff_owner.adjustFireLoss(-max(0, heal_amount - buff_owner.getBruteLoss()), passive = TRUE)
+	buff_owner.adjustBruteLoss(-heal_amount, passive = TRUE)
+	buff_owner.adjust_sunder(-1)
+	return ..()
+
+// ***************************************
+// *********** Enhancement
+// ***************************************
+/particles/drone_enhancement
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "fire"
+	color = "#9828CC"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 3
+	lifespan = 9
+	fade = 12
+	grow = 0.04
+	velocity = list(0, 0)
+	position = generator("circle", 16, 16, NORMAL_RAND)
+	drift = generator("vector", list(0, -0.15), list(0, 0.15))
+	gravity = list(0, 0.8)
+	scale = generator("vector", list(0.1, 0.1), list(0.6,0.6), NORMAL_RAND)
+	rotation = 0
+	spin = generator("num", 10, 20)
+
+/datum/status_effect/drone_enhancement
+	id = "drone enhancement"
+	duration = -1
+	tick_interval = 2 SECONDS
+	alert_type = null
+	/// Used to track the giver of this buff.
+	var/mob/living/carbon/xenomorph/buffing_target
+	/// Used to track the owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+	/// References the Essence Link action and its vars.
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action
+	/// References the Enhancement action and its vars.
+	var/datum/action/xeno_action/enhancement/enhancement_action
+	/// If the target xeno was within range.
+	var/was_within_range = TRUE
+	/// The plasma cost per tick of this ability.
+	var/plasma_cost
+
+/datum/status_effect/drone_enhancement/on_creation(mob/living/new_owner, mob/living/carbon/new_target)
+	buff_owner = new_owner
```

Comment:
You're casting from living to xenomorph without checking, same for buffing_target

Reply:
Would appreciate some clarification on this one, since I don't understand the issue.

Reply:
Asked them directly in Discord, got guidance and fixed it.